static/coffee/services.coffee

```diff
@@ -154,6 +157,19 @@ services.factory('MessageService', ['$rootScope', '$http', '$httpParamSerializer
         params.label = label.id
 
       return $http.post('/message/action/' + action + '/', params)
+
+    #----------------------------------------------------------------------------
+    # Check if message is locked, set locked state and return locked message ids
+    #----------------------------------------------------------------------------
+    checkLock: (items, unlock) ->
+      action = if unlock then 'unlock' else 'lock'
+      params = {
+        messages: if angular.isArray(items) then (i.id for i in items) else [items]
```

Comment:
It's somewhat confusing that this method takes an array of objects or a single id. Better if it always takes an array of msg objects, so I would change the JSON representation of a case to include initial_message like 
```
initial_message: {id: 123, text: "Hello"}
```

However as I've said above, I'm not sure we need to include initial message with a case but I'm not sure we need to lock that message whilst a case is being viewed.